### PR TITLE
Use gh releases

### LINKED ISSUE #281: RFE: is it possible to start making github releases?🤔
Is it possible next time on release new version make the github release to have entry on https://github.com/jbms/sphinx-immaterial/releases? 🤔

I'm asking because only on make gh release is spread notification about new release to those who have set watch->releases.
My automation process those notification trying make preliminary automated upgrade of building packages which allow save some time on maintaining packaging procedures.

More about gh releases is possible to find on
https://docs.github.com/en/repositories/releasing-projects-on-github/managing-releases-in-a-repository
https://github.com/marketplace/actions/github-release
https://pgjones.dev/blog/trusted-plublishing-2023/

### PR DESCRIPTION
**Pull request by 2bndy5**
resolves #281 

~Also auto-generates a changelog in the docs.~

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -117,6 +128,8 @@ jobs:
         with:
           macos-skip-brew-update: 'true'
       - name: Build docs
+        env:
+          SPHINX_GITHUB_CHANGELOG_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (2bndy5) [inline .github/workflows/build.yml:132]: I can only [set `permissions` of the `GITHUB_TOKEN`](https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs) per job or per workflow. Setting the token's permissions to read-only for this entire job will likely cause failure in the upload to gh-pages step.

To ensure read-only access to the changelog ext, I think A separate key may be more needed. The ext does nothing if there is no token to use.

Comment (2bndy5) [reply .github/workflows/build.yml]: I repealed the changelog generation as it can be addressed separately (if desired).

**Comment by 2bndy5** [comment]
CI requires #296 merged.

**Review by jbms — APPROVED** [review]

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -64,6 +64,17 @@ jobs:
         if: contains(matrix, 'node-version')
       - run: pipx run nox -s ${{ matrix.nox-sessions }}
 
+  create_release:
+    if: ${{ startsWith(github.ref, 'refs/tags/v') }}
+    # This needs to be done before the docs are built to ensure the latest tagged commit is
```
[the remark this answers is not in the document]
Comment (2bndy5) [reply .github/workflows/build.yml]: This comment no longer applies since I repealed the changelog generation.

**Pull request merged**